pina/label_tensor.py

```diff
@@ -5,187 +5,351 @@
-        return LabelTensor(copy_tensor, labels)
-
     @property
-    def labels(self):
-        """Property decorator for labels
-
-        :return: labels of self
-        :rtype: list
-        """
-        return self._labels
-
-    @labels.setter
-    def labels(self, labels):
-        if len(labels) != self.shape[self.ndim - 1]:  # small check
-            raise ValueError(
-                "The tensor has not the same number of columns of "
-                "the passed labels."
-            )
-
-        self._labels = labels  # assign the label
-
-    @staticmethod
-    def vstack(label_tensors):
-        """
-        Stack tensors vertically. For more details, see
-        :meth:`torch.vstack`.
-
-        :param list(LabelTensor) label_tensors: the tensors to stack. They need
-            to have equal labels.
-        :return: the stacked tensor
-        :rtype: LabelTensor
-        """
-        if len(label_tensors) == 0:
-            return []
-
-        all_labels = [label for lt in label_tensors for label in lt.labels]
-        if set(all_labels) != set(label_tensors[0].labels):
-            raise RuntimeError("The tensors to stack have different labels")
-
-        labels = label_tensors[0].labels
-        tensors = [lt.extract(labels) for lt in label_tensors]
-        return LabelTensor(torch.vstack(tensors), labels)
+    def tensor(self):
+        return self.as_subclass(Tensor)
 
-    def clone(self, *args, **kwargs):
-        """
-        Clone the LabelTensor. For more details, see
-        :meth:`torch.Tensor.clone`.
+    def __len__(self) -> int:
+        return super().__len__()
 
-        :return: A copy of the tensor.
-        :rtype: LabelTensor
-        """
-        # # used before merging
-        # try:
-        #     out = LabelTensor(super().clone(*args, **kwargs), self.labels)
-        # except:
-        #     out = super().clone(*args, **kwargs)
-        out = LabelTensor(super().clone(*args, **kwargs), self.labels)
-        return out
-
-    def to(self, *args, **kwargs):
-        """
-        Performs Tensor dtype and/or device conversion. For more details, see
-        :meth:`torch.Tensor.to`.
+    def __init__(self, x, labels):
         """
-        tmp = super().to(*args, **kwargs)
-        new = self.__class__.clone(self)
-        new.data = tmp.data
-        return new
+        Construct a `LabelTensor` by passing a dict of the labels
 
-    def select(self, *args, **kwargs):
-        """
-        Performs Tensor selection. For more details, see :meth:`torch.Tensor.select`.
+        :Example:
+            >>> from pina import LabelTensor
+            >>> tensor = LabelTensor(
+            >>>     torch.rand((2000, 3)), 
+                    {1: {"name": "space"['a', 'b', 'c'])
+        
         """
-        tmp = super().select(*args, **kwargs)
-        tmp._labels = self._labels
-        return tmp
+        from .utils import check_consistency
+        check_consistency(labels, dict)
 
```

Comment:
I don't understand completely this. Is this back-compatible? Do we need to specify for each tensor all labels for all columns or is it done automatically? What is the format?